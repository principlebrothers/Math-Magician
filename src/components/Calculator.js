/* eslint-disable react/prefer-stateless-function */
import React, { useState } from 'react';
import calculate from '../logic/calculate';

const Calculator = () => {
  const [calc, setCalc] = useState({ total: '', operation: '', next: '' });

  const handleClick = (e) => {
    setCalc(calculate(calc, e.target.innerText));
  };

  return (
    <main>
      <section className="mainPage-text">
        <h1>Let&apos;s do some Math!</h1>
      </section>
      <section className="calculator">
        <p className="display">{(calc.total || '') + (calc.operation || '') + (calc.next || '') || 0 }</p>
        <button type="button" id="AC" onClick={handleClick}>AC</button>
        <button type="button" id="+/-" onClick={handleClick}>+/-</button>
        <button type="button" id="%" onClick={handleClick}>%</button>
        <button type="button" className="operator" id="&divide;" onClick={handleClick}>&divide;</button>
        <button type="button" id="7" onClick={handleClick}>7</button>
        <button type="button" id="8" onClick={handleClick}>8</button>
        <button type="button" id="9" onClick={handleClick}>9</button>
        <button type="button" className="operator" id="x" onClick={handleClick}>x</button>
        <button type="button" id="4" onClick={handleClick}>4</button>
        <button type="button" id="5" onClick={handleClick}>5</button>
        <button type="button" id="6" onClick={handleClick}>6</button>
        <button type="button" className="operator" id="-" onClick={handleClick}>-</button>
        <button type="button" id="1" onClick={handleClick}>1</button>
        <button type="button" id="2" onClick={handleClick}> 2 </button>
        <button type="button" id="3" onClick={handleClick}> 3 </button>
        <button type="button" className="operator" id="+" onClick={handleClick}>+</button>
        <button type="button" className="zero" id="0" onClick={handleClick}>0</button>
        <button type="button" id="." onClick={handleClick}>.</button>
        <button type="button" className="operator" id="=" onClick={handleClick}>=</button>
      </section>
    </main>
  );
};

Calculator.defaultProps = { total: 0, operations: '', next: 0 };

export default Calculator;
