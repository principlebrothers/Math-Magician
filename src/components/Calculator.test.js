import React from 'react';
import {
  fireEvent, render, screen,
} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { BrowserRouter as Router } from 'react-router-dom';
import Calculator from "./Calculator";


test('should render calculator', () => {
  const calc = render(<Router><Calculator /></Router>);
  expect(calc).toMatchSnapshot();
});

test('display updates when user clicks a number', () => {
  render(<Router><Calculator /></Router>);
  expect(screen.getByText('4')).toBeInTheDocument();
  const button2 = screen.getByText('2');
  const button5 = screen.getByText('5');
  fireEvent.click(button2);
  fireEvent.click(button5);
  const display = screen.getByText('25');
  expect(display).toBeInTheDocument();
});