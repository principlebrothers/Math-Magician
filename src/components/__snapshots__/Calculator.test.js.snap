// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render calculator 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <main>
        <section
          class="mainPage-text"
        >
          <h1>
            Let's do some Math!
          </h1>
        </section>
        <section
          class="calculator"
        >
          <p
            class="display"
          >
            0
          </p>
          <button
            id="AC"
            type="button"
          >
            AC
          </button>
          <button
            id="+/-"
            type="button"
          >
            +/-
          </button>
          <button
            id="%"
            type="button"
          >
            %
          </button>
          <button
            class="operator"
            id="รท"
            type="button"
          >
            รท
          </button>
          <button
            id="7"
            type="button"
          >
            7
          </button>
          <button
            id="8"
            type="button"
          >
            8
          </button>
          <button
            id="9"
            type="button"
          >
            9
          </button>
          <button
            class="operator"
            id="x"
            type="button"
          >
            x
          </button>
          <button
            id="4"
            type="button"
          >
            4
          </button>
          <button
            id="5"
            type="button"
          >
            5
          </button>
          <button
            id="6"
            type="button"
          >
            6
          </button>
          <button
            class="operator"
            id="-"
            type="button"
          >
            -
          </button>
          <button
            id="1"
            type="button"
          >
            1
          </button>
          <button
            id="2"
            type="button"
          >
             2 
          </button>
          <button
            id="3"
            type="button"
          >
             3 
          </button>
          <button
            class="operator"
            id="+"
            type="button"
          >
            +
          </button>
          <button
            class="zero"
            id="0"
            type="button"
          >
            0
          </button>
          <button
            id="."
            type="button"
          >
            .
          </button>
          <button
            class="operator"
            id="="
            type="button"
          >
            =
          </button>
        </section>
      </main>
    </div>
  </body>,
  "container": <div>
    <main>
      <section
        class="mainPage-text"
      >
        <h1>
          Let's do some Math!
        </h1>
      </section>
      <section
        class="calculator"
      >
        <p
          class="display"
        >
          0
        </p>
        <button
          id="AC"
          type="button"
        >
          AC
        </button>
        <button
          id="+/-"
          type="button"
        >
          +/-
        </button>
        <button
          id="%"
          type="button"
        >
          %
        </button>
        <button
          class="operator"
          id="รท"
          type="button"
        >
          รท
        </button>
        <button
          id="7"
          type="button"
        >
          7
        </button>
        <button
          id="8"
          type="button"
        >
          8
        </button>
        <button
          id="9"
          type="button"
        >
          9
        </button>
        <button
          class="operator"
          id="x"
          type="button"
        >
          x
        </button>
        <button
          id="4"
          type="button"
        >
          4
        </button>
        <button
          id="5"
          type="button"
        >
          5
        </button>
        <button
          id="6"
          type="button"
        >
          6
        </button>
        <button
          class="operator"
          id="-"
          type="button"
        >
          -
        </button>
        <button
          id="1"
          type="button"
        >
          1
        </button>
        <button
          id="2"
          type="button"
        >
           2 
        </button>
        <button
          id="3"
          type="button"
        >
           3 
        </button>
        <button
          class="operator"
          id="+"
          type="button"
        >
          +
        </button>
        <button
          class="zero"
          id="0"
          type="button"
        >
          0
        </button>
        <button
          id="."
          type="button"
        >
          .
        </button>
        <button
          class="operator"
          id="="
          type="button"
        >
          =
        </button>
      </section>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
